### ISSUE TITLE
panic on graceful shutdown

### ISSUE THREAD

**Issue by tc-hib** [alexandrevicenzi/go-sse#24]
Hello, when using the `Shutdown` method, I get a panic:

> panic: send on closed channel

This is because the `removeClient` channel is closed before all HTTP handlers got closed.

I am a beginner, but to me it looks like we should never close `removeClient`.
But it will allow a goroutine leak, since goroutines trying to send on this channel will remain blocked.

I'll try to post a PR.

**Comment by alexandrevicenzi** [comment]
Share the code that you have, there are tests covering this area, it should work.

**Comment by tc-hib** [comment]
Hello, thanks for answering so quickly. In my PR the test reproduces the panic. I'm new to all this git stuff, sorry, I should have committed the test separately from the fix.

**Comment by ghost** [comment]
Hello, hope u are doing well

Is there eventually any update about this ? I am getting each time i do a ctrl+c a panic when calling shutdown like above stated

Regards

**Comment by alexandrevicenzi** [comment]
Yes, there's going to be a fix, I was working on it, but right now I don't have time to continue it.
